packages/hoc-validation-messages/src/withMessages.js

```diff
@@ -0,0 +1,119 @@
+import * as React from "react";
+import PropTypes from "prop-types";
+import Image, { badgeSizes } from "@crave/farmblocks-image";
+import Text from "@crave/farmblocks-text";
+import { fontSizes } from "@crave/farmblocks-theme";
+
+import MessageWrapper from "./styledComponents/MessageWrapper";
+import errorIconSrc from "./constants/errorIcon";
+
+export const withMessagesProps = {
+  errorIconSrc: PropTypes.string,
+  validationMessages: PropTypes.arrayOf(PropTypes.string),
+  onInvalid: PropTypes.func
+};
+
+const withMessages = WrappedComponent => {
+  return class ValidationMessages extends React.Component {
+    constructor(props) {
```

Comment:
Any reason to have a `constructor` here?

You can set the state directly in the class:
```
state = { messages: this.props.validationMessages }
```

And you don't need to bind the methods if you use arrow functions. So instead of `onInvalid(event) {`, you can use `onInvalid = event => {` and remove the binding.
Or, if you need keep these as methods, you can bind outside the constructor:
`onInvalid = this.onInvalid.bind(this);`

Ref.: https://hackernoon.com/the-constructor-is-dead-long-live-the-constructor-c10871bea599

Reply:
Yeah, no reason at all...

I'll change that, thanks